### PR TITLE
fix(ebpf): fix hidden_kernel_module error in some kernels

### PR DESCRIPTION
**Pull request by OriGlassman**
On some kernels, while iterating module_kset, we need to check if we've encountered the first module we iterated on, marking we iterated over all modules instead of continuing iterating indefinitely (max amount of iterations - causing a BPF_LOG_ID_UNSPEC error message)

### 1. Explain what the PR does

On some kernels, while iterating module_kset, we need to check if we've encountered the first module we iterated on, marking we iterated over all modules instead of continuing iterating indefinitely.

### 2. Explain how to test it
/tracee -e=hidden_kernel_module

### CODE REVIEW

File: pkg/ebpf/c/tracee.bpf.c [hunk 1]
```diff
@@ -770,6 +772,17 @@ statfunc bool find_modules_from_module_kset_list(program_data_t *p)
         if (mod_kobj) {
             struct module *mod = BPF_CORE_READ(mod_kobj, mod);
             if (mod) {
+                if (first_mod_name == NULL) {
```
Comment (AlonZivony) [inline pkg/ebpf/c/tracee.bpf.c:775]: Can't you just compare the module address?
Why do you need the address of the name?

Comment (OriGlassman) [reply pkg/ebpf/c/tracee.bpf.c]: Totally right. Was too deep with the kobjects. Thanks!

**Review by rafaeldtinoco — APPROVED** [review]
LGTM

**Comment by rafaeldtinoco** [comment]
Is this solving the recently reported issue (from an user) where IR code wasn't unrolling this loop more than 60 times (other than the 600 default) ? @AlonZivony @OriGlassman

**Comment by OriGlassman** [comment]
> Is this solving the recently reported issue (from an user) where IR code wasn't unrolling this loop more than 60 times (other than the 600 default) ?

Not aware of such issue, can you provide a link?

**Comment by rafaeldtinoco** [comment]
> @rafaeldtinoco
> 
> > Is this solving the recently reported issue (from an user) where IR code wasn't unrolling this loop more than 60 times (other than the 600 default) ?
> 
> Not aware of such issue, can you provide a link?

https://aquasecurity.slack.com/archives/C03MEDREVL7/p1703659096759989

**Pull request merged**